sailfish-rbc/src/abraham/worker.rs

```diff
@@ -950,15 +963,24 @@ impl<C: RawComm, T: Clone + Committable + Serialize + DeserializeOwned> Worker<C
         trace!(node = %self.label, "retrying ...");
         // Go over RBC messages and check status:
         for (digest, tracker) in self.buffer.values_mut().flat_map(|m| m.map.iter_mut()) {
-            trace!(
-                node    = %self.label,
-                digest  = %digest,
-                status  = %tracker.status,
-                retries = %tracker.retries,
-                "revisiting"
+            if matches!(tracker.status, Status::Init | Status::Delivered) {
+                continue; // nothing to do here (1).
+            }
+            debug!(
+                node     = %self.label,
+                digest   = %digest,
+                present  = %tracker.message.item.is_some(),
+                status   = %tracker.status,
+                ours     = %tracker.ours,
+                elapsed  = ?tracker.start.elapsed(),
+                retries  = %tracker.retries,
+                votes    = %tracker.votes.votes(&digest.commit()),
+                "revisiting ..."
```

Comment:
Could this perhaps be more clear? At first glance since I haven't touched this yet is that I'm not sure what's going on.

Reply:
This is near the top of the `retry` method which is invoked with a fixed frequency and revisits all buffered values. What message would you like to see here?